<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="use_nodelet" default="true"/>
  <arg name="manager_name" default="nodelet_manager"/>
  <arg name="input_points_topic" default="points_in"/>
  <arg name="output_points_topic" default="points_out"/>
  <arg name="publish_image" default="false"/>
  <arg name="resolution" default="1.0"/>
  <arg name="vfov" default="30.0"/>
  <arg name="hfov" default="360.0"/>
  <arg name="surface_threshold" default="0.1"/>
  <arg name="corner_threshold" default="1.0"/>

  <group if="$(arg use_nodelet)">
    <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen"/>

    <node pkg="nodelet" type="nodelet" name="pcd_process_loader" output="screen" args="load line_rot_slam/PcdProcessNodelet $(arg manager_name)">
      <param name="input_points_topic" value="$(arg input_points_topic)"/>
      <param name="output_points_topic" value="$(arg output_points_topic)"/>
      <param name="publish_image" value="$(arg publish_image)"/>
      <param name="resolution" value="$(arg resolution)"/>
      <param name="vfov" value="$(arg vfov)"/>
      <param name="hfov" value="$(arg hfov)"/>
      <param name="surface_threshold" value="$(arg surface_threshold)"/>
      <param name="corner_threshold" value="$(arg corner_threshold)"/>
    </node>
  </group>

  <group unless="$(arg use_nodelet)">
    <node pkg="nodelet" type="nodelet" name="pcd_process_loader" output="screen" args="standalone line_rot_slam/PcdProcessNodelet">
      <param name="input_points_topic" value="$(arg input_points_topic)"/>
      <param name="output_points_topic" value="$(arg output_points_topic)"/>
      <param name="publish_image" value="$(arg publish_image)"/>
      <param name="resolution" value="$(arg resolution)"/>
      <param name="vfov" value="$(arg vfov)"/>
      <param name="hfov" value="$(arg hfov)"/>
      <param name="surface_threshold" value="$(arg surface_threshold)"/>
      <param name="corner_threshold" value="$(arg corner_threshold)"/>
    </node>
  </group>
</launch>
